client/tinychain/collection/tensor.py

```diff
@@ -632,6 +635,52 @@ def gradients(self, loss):
         return Gradients()
 
 
+class Reduce(Dual):
+    # TODO: move common functionality from reduce operators into this parent class
+    pass
+
+
+# TODO: make it work for multiple functions and for .sum(axis=tuple)
+class Sum(Reduce):
+    def forward(self):
+        return NDArray.sum(self.subject, axis=self.args)
+
+    def backward(self, variable=None):
+        from tinychain.collection.tensor import Dense
+
+        return Dense.ones_like(self.subject)
+
+    def gradients(self, loss):
+        if self.args is None:
+            loss = self.backward() * loss
+        else:
+            axis = self.args
+            if isinstance(self.subject.shape, Tensor):
```

Comment:
I added a helper method to compute the reduced shape from the original shape in commit 962e792